### ISSUE TITLE
JRuby 9.0.5.0: %Z behavior is incompatible between strptime and strftime

### ISSUE THREAD

**Issue by frsyuki** [jruby/jruby#3702]
And its behavior is different from CRuby.
### Environment

On Mac OS X, using System Preference.app, I set system timezone to JST (+0900).
### Actual Behavior

JRuby 9.0.5.0:

```
irb(main):003:0> require 'time'
=> true
irb(main):002:0> Time.strptime('2016-08-01 00:00:00 PDT', '%Y-%m-%d %H:%M:%S %Z').strftime('%Y-%m-%d %H:%M:%S %Z')
=> "2016-08-01 00:00:00 UTC+7"
irb(main):003:0> Time.strptime('2016-08-01 00:00:00 UTC+7', '%Y-%m-%d %H:%M:%S %Z').strftime('%Y-%m-%d %H:%M:%S %Z')
=> "2016-08-01 00:00:00 JST"
irb(main):004:0> Time.strptime('2016-08-01 00:00:00 UTC+7', '%Y-%m-%d %H:%M:%S %z').strftime('%Y-%m-%d %H:%M:%S %Z')
=> "2016-08-01 00:00:00 JST"
```

JRuby's `strftime` with `%Z` outputs `UTC+7`, and `strptime` doesn't understand it (with neither `%z` or `%Z`).

CRuby 2.3.0:

```
irb(main):028:0> require 'time'
=> false
irb(main):029:0> Time.strptime('2016-08-01 00:00:00 PDT', '%Y-%m-%d %H:%M:%S %Z').strftime('%Y-%m-%d %H:%M:%S %Z')
=> "2016-08-01 00:00:00 "
irb(main):030:0> Time.strptime('2016-08-01 00:00:00 UTC+7', '%Y-%m-%d %H:%M:%S %Z').strftime('%Y-%m-%d %H:%M:%S %Z')
=> "2016-08-01 00:00:00 JST"
irb(main):031:0> Time.strptime('2016-08-01 00:00:00 UTC+7', '%Y-%m-%d %H:%M:%S %z').strftime('%Y-%m-%d %H:%M:%S %Z')
=> "2016-08-01 00:00:00 JST"
```

CRuby's `strftime` with `%Z` outputs ` `(one space), and of course,`strptime` doesn't understand it.
### Expected Behavior
- a) JRuby 9.0.5.0 behaves exactly same with CRuby, or
- b) JRuby 9.0.5.0's `strptime` can parse `UTC+7` using `%Z%` (and improve CRuby as well), or
- c) JRuby 9.0.5.0's `strftime` outputs `PDT` or `+0700` using `%Z%` (and improve CRuby as well)

Although it's behavior is different from CRuby, JRuby 9.0.5.0's improvement is good, I think. So, I prefer b) or c).

**Comment by nurse** [comment]
In this case, it's not a matter of strftime. Because the receiver, created by strptime, doesn't have timezone name.

So this is an issue of strptime, "How should strptime handle %Z?". The string is system/local dependent. For example Ruby on Windows, it is "Pacific Standard Time" or "日本標準時". Maybe it can be used for display, it can't be used for timezone information. glibc says [simple answer](https://github.com/lattera/glibc/blob/59ba27a63ada3f46b71ec99a314dfac5a38ad6d2/time/strptime_l.c#L747) for this.

People must get the offset with "%z".

**Comment by frsyuki** [comment]
@nurse do you mean that applications shouldn't use starptime with %Z? Then, Expected Behavior "a)" sounds the right approach because actual behavior doesn't matter.

**Comment by nurse** [comment]
Yeah, application shouldn't use %Z. They must get time offset or IANA timezone name, which can get from %z or ... something else.

I don't have any opinion about JRuby's strptime(%Z).

**Comment by maclover7** [comment]
Should this issue be closed since #3758 was merged in?

**Comment by kares** [comment]
was fixed for 9.1.0.0 to match up with MRI
